### PR TITLE
Fix MISRA_C_2012 rule 13.2 violation

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Fix MISRA C 2012 rule 13.2 violation

Description
-----------
**Misra Rule 13.2**

> The value of an expression and its persistent side effects shall be the same under all permitted evaluation orders.

**MISRA violation**
MISRA states that there shall be no more than one read access with volatile-qualified type within one sequence point.
The justification for the rule is:
If more than one volatile-qualified variable is read between sequence points, then it is unspecified which is read first. Reading a volatile variable is a side effect.

**Fix**
The solution is to explicitly order the reads, using a new set of static variables.

Test Steps
-----------
N/A.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tasks.c [hunk 1]
```diff
@@ -4477,13 +4479,15 @@ char * pcTaskGetName( TaskHandle_t xTaskToQuery ) /*lint !e971 Unqualified char
...
         /* Correct the tick count value after a period during which the tick
          * was suppressed.  Note this does *not* call the tick hook function for
          * each stepped tick. */
-        configASSERT( ( xTickCount + xTicksToJump ) <= xNextTaskUnblockTime );
+        configASSERT( ( xConstTickCount + xTicksToJump ) <= xConstNextTaskUnblockTime );
 
-        if( ( xTickCount + xTicksToJump ) == xNextTaskUnblockTime )
+        if( ( xConstTickCount + xTicksToJump ) == xConstNextTaskUnblockTime )
         {
             /* Arrange for xTickCount to reach xNextTaskUnblockTime in
```
Comment (ActoryOu) [inline tasks.c:4492]:
```suggestion
            /* Arrange for xConstTickCount to reach xNextTaskUnblockTime in
```

**Review by ActoryOu — COMMENTED** [review]
One minor comment. Please take a look.

File: tasks.c (hunk 1 above)
Comment (kar-rahul-aws) [reply tasks.c]: updated.

**Review by ActoryOu — DISMISSED** [review]

**Review by ActoryOu — APPROVED** [review]

File: tasks.c [hunk 2]
```diff
@@ -3854,7 +3856,7 @@ void vTaskSuspendAll( void )
         }
         else
         {
-            xReturn = xNextTaskUnblockTime - xTickCount;
+            xReturn = xConstNextTaskUnblockTime - xConstTickCount;
```
Comment (chinglee-iot) [inline tasks.c:3859]: Maybe xConstTickCount is enough to get ride of access multiple voilate variable in one statement. For example,
```c
xReturn = xNextTaskUnblockTime - xConstTickCount;
```

File: tasks.c [hunk 3]
```diff
@@ -8366,14 +8373,14 @@ static void prvAddCurrentTaskToDelayedList( TickType_t xTicksToWait,
                 /* Wake time has overflowed.  Place this item in the overflow
                  * list. */
                 traceMOVED_TASK_TO_OVERFLOW_DELAYED_LIST();
-                vListInsert( pxOverflowDelayedTaskList, &( pxCurrentTCB->xStateListItem ) );
+                vListInsert( pxOverflowDelayedList, &( pxCurrentTCB->xStateListItem ) );
```
Comment (chinglee-iot) [inline tasks.c:8376]: `prvAddCurrentTaskToDelayedList` is called with scheduler suspended. Can we consider the following code to reduce the logic?
```
ListItem_t * pxCurrentTaskStateItem = pxCurrentTCB->xStateListItem;

vListInsert( pxDelayedTaskList, pxCurrentTaskStateItem );
vListInsert( pxOverflowDelayedList, pxCurrentTaskStateItem );
```

File: tasks.c (hunk 2 above)
Comment (kar-rahul-aws) [reply tasks.c:3859]: Updated.

File: tasks.c (hunk 3 above)
Comment (kar-rahul-aws) [reply tasks.c:8376]: This does not impact the MISRA violation, do you want this in a separate PR?

Comment (chinglee-iot) [reply tasks.c:8374]: The comment above is suggestion only. The original implementation is okay to me.

**Review by chinglee-iot — APPROVED** [review]

File: tasks.c (hunk 3 above)
Comment (kar-rahul-aws) [reply tasks.c:8374]: Ok.

**Review by moninom1 — DISMISSED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**